src/mon/MonitorDBStore.h

```diff
@@ -652,10 +659,56 @@ class MonitorDBStore
       db->init(g_conf()->mon_rocksdb_options);
     else
       db->init();
+    return 0;
+  }
 
-
+  int _lock() {
+    assert(lock_fd < 0);
+    char fn[4096];
+    snprintf(fn, sizeof(fn)-1, "%s/kv_backend", path.c_str());
+    lock_fd = ::open(fn, O_RDWR);
+    if (lock_fd < 0) {
+      generic_derr << __func__ << " unable to open " << fn << " for lock" << dendl;
+      return -errno;
+    }
+
+    uint64_t nr_tries = 10;
```

Comment:
i don't think we should retry in this case. the reason why i wanted to retry in `KernelDevice::_lock()` was to wait for `systemd-udevd` until it relinquish the advisory lock.

but in this case, the only holder of the lock is another monitor process, so, instead of waiting for it, we should just bail out with an error message.